In [7]:
import pandas as pd
import matplotlib.pyplot as plt



### SAMPEL DATA

Jika kita lihat dari 10 data pertama yang ada dalam dataa set maka kita hanya akan mendapatkan bahwa dari tahun 1980 sampai dengan 1989 pengguna internet di Afganistan masih lah 0

In [8]:
df=pd.read_csv(r"C:\Users\HP\Downloads\datasetTUBES.csv")
df.loc[df["Entity"]=="Afghanistan"].head(5)

Unnamed: 0       Entity Code Continent  Year  No. of Internet Users  \
0           0  Afghanistan  AFG      Asia  1980                      0   
1           1  Afghanistan  AFG      Asia  1981                      0   
2           2  Afghanistan  AFG      Asia  1982                      0   
3           3  Afghanistan  AFG      Asia  1983                      0   
4           4  Afghanistan  AFG      Asia  1984                      0   

   Internet Users(%)  Cellular Subscription  Broadband Subscription  
0                0.0                    0.0                     0.0  
1                0.0                    0.0                     0.0  
2                0.0                    0.0                     0.0  
3                0.0                    0.0                     0.0  
4                0.0                    0.0                     0.0

In [43]:

df.loc[df["Entity"]=='Afghanistan'].loc[(df["Year"]>=2000) & (df["Year"]<=2003)]
#Berdasarkan data dibawah dapat kita lihat bahwa untuk negara Afganistan pengguna internet 
# mulai bertambah dimulai dari tahun 2001 


Unnamed: 0       Entity Code Continent  Year  No. of Internet Users  \
20          20  Afghanistan  AFG      Asia  2000                      0   
21          21  Afghanistan  AFG      Asia  2001                    930   
22          22  Afghanistan  AFG      Asia  2002                    958   
23          23  Afghanistan  AFG      Asia  2003                  19903   

    Internet Users(%)  Cellular Subscription  Broadband Subscription  
20           0.000000               0.000000                     0.0  
21           0.004723               0.000000                     0.0  
22           0.004561               0.110616                     0.0  
23           0.087891               0.844564                     0.0

Berdasarkan data di atas dapat kita lihat bahwa untuk negara Afganistan pengguna internet mulai bertambah dimulai dari tahun 2001 

In [44]:
df.loc[df["Entity"]=="Afghanistan"].tail(5)

Unnamed: 0       Entity Code Continent  Year  No. of Internet Users  \
36          36  Afghanistan  AFG      Asia  2016                3809983   
37          37  Afghanistan  AFG      Asia  2017                4811862   
38          38  Afghanistan  AFG      Asia  2018                6163380   
39          39  Afghanistan  AFG      Asia  2019                6647431   
40          40  Afghanistan  AFG      Asia  2020                7170891   

    Internet Users(%)  Cellular Subscription  Broadband Subscription  
36          11.000000              61.054638                0.024874  
37          13.500000              65.929131                0.025297  
38          16.799999              59.120850                0.043041  
39          17.600000              59.356018                0.051741  
40          18.400000              58.255814                0.068254

Dari 5 data terakhir pada negara Afghanistan tersebut dapat kita ketahui bahwa semakin besar tahunnya maka jumlah dari pengguna internet di Afganistan juga akan bertambah

In [45]:
a=df.loc[(df["Entity"]!= "World")].loc[(df["Continent"]!= "Undefined")]
a.sort_values("No. of Internet Users", ascending=False).head(1)





Unnamed: 0 Entity Code Continent  Year  No. of Internet Users  \
1352        1619  China  CHN      Asia  2020             1003218650   

      Internet Users(%)  Cellular Subscription  Broadband Subscription  
1352          70.404778              119.39016               33.595604

Berdasaarkan informasi tersebut dapat dilihat bahwa negara china memiliki Jumlah pengguna internet yang paling banyak dengan persentase pengguna internet di nengara itu mencapai 70% pada tahun 2020

In [46]:
df.sort_values(["Internet Users(%)"], ascending=[False]).head(1)


Unnamed: 0   Entity Code Continent  Year  No. of Internet Users  \
449         580  Bahrain  BHR      Asia  2019                1489735   

     Internet Users(%)  Cellular Subscription  Broadband Subscription  
449          99.701492             115.792923                8.590934

Berdasarkan data tersebut kita dapat mengetahui bahwa walaupun jumlah pengguna internetnya banyak belum tentu semua orang di negara tersebut merupakan pengguna internet. Negara yang mayoritas masyarakatnya sudah menggunakan internet adalah negara Bahrian dengan persentase sampai 99,7%

### STATISTIK

In [15]:
a=df.loc[df["Entity"]=="World"]
a.describe(percentiles=[0.1, 0.25, 0.5, 0.75, 0.9]).drop(columns='Year')


No. of Internet Users  Internet Users(%)  Cellular Subscription  \
count           4.100000e+01          41.000000              41.000000   
mean            1.107868e+09          15.308309              34.819115   
std             1.390404e+09          18.322288              41.270557   
min             0.000000e+00           0.000000               0.000530   
10%             0.000000e+00           0.000000               0.006855   
25%             2.617438e+06           0.049235               0.210930   
50%             4.140127e+08           6.733119              12.042665   
75%             2.021153e+09          28.933121              76.161530   
90%             3.256216e+09          43.338352             100.720001   
max             4.699886e+09          59.940247             106.561058   

       Broadband Subscription  
count               41.000000  
mean                 3.984538  
std                  5.268207  
min                  0.000000  
10%                  0.000000  
25%                  0.000000  
50%                  0.000000  
75%                  7.848253  
90%                 12.431141  
max                 15.887396

In [5]:
countries_2020= df[(df["Continent"]!="Undefined") &(df["Year"]==2020) & (df["No. of Internet Users"]!=0)] 
countries_2020.describe(percentiles=[0.1, 0.25, 0.5, 0.75, 0.9]).drop(columns='Year')

Unnamed: 0  No. of Internet Users  Internet Users(%)  \
count   126.000000           1.260000e+02         126.000000   
mean   4180.222222           3.079839e+07          63.964468   
std    2573.880448           1.059077e+08          25.449012   
min      40.000000           4.806000e+04           9.400000   
10%     786.000000           6.093935e+05          25.400000   
25%    1874.250000           2.061975e+06          40.849999   
50%    4050.000000           5.719996e+06          70.444103   
75%    6357.500000           1.872981e+07          85.066164   
90%    7767.500000           6.446629e+07          92.084946   
max    8866.000000           1.003219e+09          99.669998   

       Cellular Subscription  Broadband Subscription  
count             126.000000              126.000000  
mean              115.051087               16.398783  
std                43.918233               15.025334  
min                37.910732                0.000000  
10%                67.232227                0.230081  
25%                95.228342                1.794761  
50%               114.002125               12.408129  
75%               132.368916               30.325385  
90%               148.942055               39.374537  
max               430.132355               48.334816

In [6]:

buru=df.iloc[countries_2020["Internet Users(%)"].idxmin()]
bahri=df.iloc[countries_2020["Internet Users(%)"].idxmax()]
result=pd.concat([buru,bahri], axis=1)
result = result.rename(columns={1024: 'MIN', 450: 'MAX'}).drop('Unnamed: 0',axis=0)
result

MIN         MAX
Entity                    Burundi     Bahrain
Code                          BDI         BHR
Continent                  Africa        Asia
Year                         2020        2020
No. of Internet Users     1148701     1472602
Internet Users(%)             9.4   99.669998
Cellular Subscription   55.767147  102.767845
Broadband Subscription   0.035574    8.752362

Berdasarkan data yang di atas, perbedaan persentase pengguna internet antara Burundi dan Bahrain sangat signifikan.

1.  **Burundi:** Persentase pengguna internet hanya sekitar 9.4%. Angka ini menunjukkan bahwa jumlah penduduk yang menggunakan internet di negara ini relatif rendah pada tahun 2020.
    
2.  **Bahrain:** Sementara di Bahrain, persentase pengguna internet sangat tinggi, mencapai sekitar 99.67%. Angka ini menandakan bahwa hampir semua penduduk di Bahrain telah mengakses internet pada tahun tersebut.
    

Perbedaan ini menunjukkan kesenjangan yang besar dalam akses dan penetrasi internet antara kedua negara tersebut. Bahrain memiliki tingkat penetrasi internet yang hampir mendekati keseluruhan populasi, sementara Burundi masih memiliki tingkat akses yang jauh lebih rendah.

Kesenjangan seperti ini dapat memiliki dampak besar terutama dalam hal akses informasi, komunikasi, kesempatan pendidikan, dan pertumbuhan ekonomi karena akses yang terbatas terhadap internet dapat membatasi akses terhadap informasi dan kesempatan lainnya.

Untuk meningkatkan persentase pengguna internet di Burundi, pemerintah dapat fokus pada beberapa langkah strategis. Investasi dalam infrastruktur telekomunikasi yang lebih luas dan stabil diperlukan, terutama di daerah pedesaan. Kebijakan pro-internet serta insentif untuk perusahaan telekomunikasi bisa mendorong pertumbuhan ekosistem digital. Program pendidikan dan literasi digital akan memperkuat pemahaman tentang manfaat internet. Kolaborasi dengan organisasi internasional dan swasta, serta memastikan akses internet terjangkau, juga akan menjadi kunci dalam memperluas akses internet bagi penduduk. Inovasi teknologi lokal juga dapat membantu menciptakan solusi yang lebih terjangkau dan efektif untuk meningkatkan akses internet di negara tersebut.